concurrency: add counting thread app

davidkostov opened the pull request:
Added two threads application, 1 thread counts while the other waits for a specific string to output the current value of the counter from the first thread.

mgenov commented on src/main/java/com/clouway/threads/Main.java:
```diff
@@ -0,0 +1,19 @@
+ */
+public class Main {
+
+    public static void main(String[] args) {
```
Doesn't look correct If this is task I
>След стартирането на нишката, в главната прорама да се чака въвеждането на символ от клавиатурата. След прочитането на символа да се спира нишката и да се извежда до коя стойност е стигнало отброяването.

The idea of this task is to learn you how to ask another thread to terminate it's execution. 

https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html

davidkostov commented:
Sorry.. I missed the part that says the thread HAS to be interrupted. I fixed it, also removed listener since the task says everything should happen in the main.

mgenov commented on src/main/java/com/clouway/threads/Counter.java:
```diff
@@ -0,0 +1,24 @@
+
+    private Integer count;
+
+    public synchronized Integer get() {
```
Counter could print it's count on it's own and this synchronization won't be required.
